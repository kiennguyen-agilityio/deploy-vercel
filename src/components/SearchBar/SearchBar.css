.search-bar-container {
  position: relative;
  display: inline-block;
  width: 627px;
  height: 60px;
  border: 1px solid var(--color-black);
  border-radius: 10px;
}

.search-bar {
  font-family: 'Open Sans', sans-serif;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  padding-left: 25px;
  padding-right: 25px;
  border: none;
  background-color: transparent;
}

.search-icon {
  position: absolute;
  top: 50%;
  right: 25px;
  transform: translateY(-50%);
  font-size: var(--font-size-regular);
  color: var(--color-black);
}

.fa-sharp {
  font-size: var(--font-size-md);
  color: var(--color-black);
}
