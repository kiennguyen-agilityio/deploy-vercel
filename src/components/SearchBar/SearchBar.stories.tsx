import { StoryObj, Meta } from '@storybook/react';
import SearchBar from './SearchBar';

export default {
  title: 'Components/SearchBar',
  component: SearchBar,
} as Meta;

type Story = StoryObj<typeof SearchBar>;

export const Default: Story = {
  args: {
    searchValue: '',
    onSetSearchValue: (value) => console.log(value),
  },
};
