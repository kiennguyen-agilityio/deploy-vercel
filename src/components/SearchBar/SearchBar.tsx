import { memo } from 'react';

// style
import './SearchBar.css';

interface Props {
  searchValue: string;
  onSetSearchValue: (value: string) => void;
}

const SearchBar: React.FC<Props> = ({ searchValue, onSetSearchValue }) => {
  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>): void => {
    onSetSearchValue(event.target.value);
  };

  return (
    <div className="search-bar-container">
      <input className="search-bar" onChange={handleSearch} value={searchValue} />
      <i className="search-icon fa-sharp fa-solid fa-search" />
    </div>
  );
};

export default memo(SearchBar);
