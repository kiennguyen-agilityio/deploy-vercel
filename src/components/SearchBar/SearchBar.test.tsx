import { render } from '@testing-library/react';
import SearchBar from './SearchBar';

test('renders without crashing', () => {
  const { container } = render(<SearchBar searchValue="" onSetSearchValue={() => {}} />);
  expect(container.firstChild).toMatchSnapshot();
});
