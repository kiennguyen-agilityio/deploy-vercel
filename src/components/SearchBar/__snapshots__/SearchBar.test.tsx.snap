// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="search-bar-container"
>
  <input
    class="search-bar"
    value=""
  />
  <i
    class="search-icon fa-sharp fa-solid fa-search"
  />
</div>
`;
